Sync v2-4-stable with v2-4-test to release 2.4.3

ephraimbuddy opened the pull request:
Time for `2.4.3rc1`!

ashb committed:
Respect "common" options value in breeze sub-commands. (#26264)

This lets us run `breeze -v static-checks` and have the sub-command pick
up the value from the parent context. If you specify a value for the
subcommand, that is used in place of the global one.

(The fact that `-v` was allowed before the sub-command but had no effect
caused me lots of confusion)

(cherry picked from commit 6da571f705bbfd92f8641f2604ef6f4c61bfa889)

potiuk committed:
Update CI documentation, renaming runs to "Canary" (#26151)

For quite some time we did not have the name of the main builds
However more and more the "main" builds are used to provide early
warnigns for some problems:

* 3rd-party dependencies breaking our builds
* our own dependencies breaking the constraints
* building ARM images
* building breeze images quickly
* running complete matrix of tests
* finding flaky tests

So effectively, those main builds are really "Canary" builds - when
those builds are failing, they give us a chance to react quickly,
without affecting the regular PR builds.

This PR clarifies the meaning and reasoning for those builds
and introduces "Canary" name for them.

During related documentation review, it also turned out that a number
of old environment variables are not used any more (after the breeze
changing to Python) and this PR also removes them from documentation
and removes the variables from all the scripts (including removal of
some unused scripts)

The new documentation also mentions somethign that we've learned
recently - that in case you use Breeze in non-airflow workflows in
GitHub Actions, you need to override the variables through command
line parameters rather than through environment variables, because
GitHub actions treats GITHUB_* variables as immutable for security.

(cherry picked from commit eb03959e437e11891b8c3696b76f664a991a37a4)

norm committed:
Create a workflow to sync the list of authors (#26275)

The CI jobs caused by Pull Requests run on Airflow infrastructure only
when an author is trusted, which currently means a core committer. The
list is currently duplicated in the ci.yml workflow.

As part of being able to open up the infra to other trusted authors,
this list needs to be updated based on a list of trusted authors kept
in the `airflow-ci-infra` repository.

(cherry picked from commit 73257628219e09c6cfd14b38d0aeed8b6e46a83c)

potiuk committed:
Fix static checks from sync_authors script (#26365)

The #26275 introduced static check problem (isort). This PR fixes it

(cherry picked from commit aa0118b0d2a4085f80663b14c1d58b54cab81319)

potiuk committed:
Actually run the "quick image build" (#26351)

The check does not use matrix build but we left the check for
matrix so the check was always skipped.

(cherry picked from commit c89e92f63aca7b1b008604883ce08c0ce032c80b)

norm committed:
Use the full path to the workflow being edited (#26386)

(cherry picked from commit 3e917f1c02c4654a135c9a740ee264b766bbc58a)

gmsantos committed:
Fix params reference for default_constraints_branch (#26406)

Fixes breeze commands that depends on the default constraints branch,
such as the generated diff for
`breeze release-management generate-constraints`

(cherry picked from commit e458eb689d70e4857cdc3cdaaaafd4bfdbcc8ed7)

potiuk committed:
Attempt to create k8s venv without constraints if constraint build fails (#26407)

There is an issue with trying to install the k8s env where the
current requirements conflict with constraints. The k8s env
creation is failing in this case.

This PR will not fail hard when the constraint-build fails, instead
it will attempt to build the venv without constraints. This will work
in vast majority of cases.

(cherry picked from commit 6f390fa3c2b5d56de1454f1ba802a7b81853142e)

norm committed:
Credentials are needed in order to push branch (#26413)

(cherry picked from commit 9115488a68f7c3112f0c4943c32d318a8222cf96)

norm committed:
Manual triggering of author sync workflow (#26414)

* Allow committers to run author sync manually

* Fix test failures

(cherry picked from commit 4261210aecbac43fba355d348beb3301f4f79ac5)

potiuk committed:
Fix problem with building providers where constraints are conflicting (#26420)

When constraints are conflicting with Airflow's requirement,
the provider build fails. This change makes it retry the installation
without constraints in such case.

(cherry picked from commit 954a5510e5324cb78c9b386c750eb4168f1c8f53)

norm committed:
Add workflows permission (#26416)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 2382c12cc3aa5d819fd089c73e62f8849a567a0a)

ashb committed:
Revert "Add workflows permission (#26416)" (#26422)

This reverts commit 2382c12cc3aa5d819fd089c73e62f8849a567a0a.

(cherry picked from commit 7a3a752c64a0f96f0627f08061e01cbc71494df4)

potiuk committed:
Improve 'start-airflow' experience for users and ui developers (#26341)

Since we moved the "compile-www-assets" to host (and deferred it
to pre-commit to prepare and manage appropriate node virtualenv),
starting airflow via `start-airflow` command had become slower,
adding quite a few seconds always when you run the command. Also
it was a bit convoluted to start dev mode - you had to stop
and restart webserver and run dev asset recompilation.

This change improves the `start-airflow` experience in several ways:

* the compile-www-assets pre-commit in start airflow runs in a
  background, daemon thread. This allows start-airflow command
  to continue even if asset compilation takes quite some time.
  By the time webserver starts and database initializes and the user
  launches browser it should be completed anyway.
* the assets are not recompiled if none of the "www" files changed which
  for most people who run airflow without modifying the UI part
  will make it really fast to re-run start-airflow
* we have a new flag --skip-assets-compilation added that will skip
  asset compilation even if the www files were modified
* finally we have a new "--dev-mode" when starting airflow that will
  run the "dev" mode of assets compilation. And thanks to the
  background, daemon thread, assets compilation in dev mode will
  continue running as long as the `start-airflow` command is running

All this should naturally plug in the exisitng start-airflow usage
and add new capabilities for those who develop UI (you can run
dev mode now without manually restarting anything).

(cherry picked from commit 7661f34869e44e9550455d0411c127889acb6b34)

jedcunningham committed:
Fix www asset compilation (#26445)

(cherry picked from commit f600197ad47916687830177c9787c4616f7854a2)

potiuk committed:
Switch back breeze image regeneration to be non-docker (#26458)

* Switch back breeze image regeneration to be non-docker

The change to move image regeneration to docker has been
dictated by slightly inconsistent behaviour on various
deployment. The idea was that Docker will solve it but ut has not
turned out to be true.

Generating images wihtout the need of image is much faster and more
portable and we need to look for some other ways of improving
the consistency.

* Update dev/breeze/src/airflow_breeze/commands/setup_commands.py

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>
(cherry picked from commit 23e6cf2c447c3ff6a7057daffcbf04bfd96b9cf9)

kokorin committed:
Fix: git clone on Windows (#26105)

(cherry picked from commit 9aa589ec90a6e56837495334f5eadf62158274e4)

potiuk committed:
Fix partial regeneration of breeze hash commands on pre-commit (#26550)

Pre-commit moved to Host in #26458 caused only partial
dump for hash file (with main and cleanup method) due to the
way breeze commands were initialized by click. We are now running
actual breeze command to regenerate those commands so they
should always contain all commands.

(cherry picked from commit af2476f29d01376c2f66ed0cff57895f6a9444e3)

ashb committed:
Ensure we don't include environment specific defaults in help images (#26552)

With the switch to generating the breeze images in the host, not in
docker we changed how we invoke the sub-commands, so the previous method
of detecting when we are in image generation mode via `sys.argv` no
longer works. Instead we now look for the existing environment var we
set to configure "recording" mode.

(cherry picked from commit e9b026d64e22c06b6ba2dff3685910aa18383873)

ashb committed:
Pull in licneses to dists via wildcard, not pre-commit (#26553)

We can simplify the build/pre-commit process a bit. It turns out that
`license_files` is actually a list of glob-patterns, not just literal
files, so we can remove the pre-commit step to generate this list and
keep it ordered.

Test to make sure it still gets included:

```
$unzip -l dist/apache_airflow-2.5.0.dev0-py3-none-any.whl | grep apache_airflow-2.5.0.dev0.dist-info
    13661  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE
     1131  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-bootstrap.txt
      984  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-bootstrap3-typeahead.txt
     1475  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-d3-shape.txt
     1079  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-d3-tip.txt
     1475  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-d3js.txt
     1062  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-dagre-d3.txt
     1076  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-datatables.txt
     1081  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-elasticmock.txt
     1096  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-eonasdan-bootstrap-datetimepicker.txt
     1301  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-flask-kerberos.txt
    11349  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-hue.txt
    11357  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-jqclock.txt
     1605  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-jquery.txt
     1077  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-moment-strftime.txt
     1075  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-moment.txt
     1096  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSE-normalize.txt
   115987  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/LICENSES-ui.txt
   120239  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/METADATA
      789  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/NOTICE
       92  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/WHEEL
       50  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/entry_points.txt
        8  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/top_level.txt
    93366  2022-09-21 11:00   apache_airflow-2.5.0.dev0.dist-info/RECORD
$  tar tf dist/apache-airflow-2.5.0.dev0.tar.gz apache-airflow-2.5.0.dev0/licenses/
apache-airflow-2.5.0.dev0/licenses/
apache-airflow-2.5.0.dev0/licenses/LICENSE-bootstrap.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-bootstrap3-typeahead.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-d3-shape.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-d3-tip.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-d3js.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-dagre-d3.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-datatables.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-elasticmock.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-eonasdan-bootstrap-datetimepicker.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-flask-kerberos.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-hue.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-jqclock.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-jquery.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-moment-strftime.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-moment.txt
apache-airflow-2.5.0.dev0/licenses/LICENSE-normalize.txt
apache-airflow-2.5.0.dev0/licenses/LICENSES-ui.txt
```

(cherry picked from commit ba9a762c71b9edae3948e67a4bfc97c93cf5c15e)

potiuk committed:
Add back checking image help environment disable on sys.argv (#26562)

(cherry picked from commit 520893a9a13a6ec5611ed4dd76039974eb4c069c)

potiuk committed:
Switch to reading package list from provider_dependencies.json (#26572)

The list of packages in breeze for building docs have been built by
scanning the "docs" folder - but the content of it was somewhat
depending on which folders were available there - and they could
have been generated / added locally by independently developed PRs
and left-over.

This PR solves it by replacing the source of packages to be
provider_dependencies.json (+ few hard-coded packages).

This will make build-docs only rely on committed sources.

(cherry picked from commit 66a441cff2b375450bd487af502dcf5067854679)

dstandish committed:
Make sure ast has attr unparse in pre-commit (#26578)

(cherry picked from commit 86e7f3af26512c36f19ae52f709e639f47bdb168)

ashb committed:
Convert release manager instructions to use "partial" svn checkouts (#26589)

This just makes it a bit easier to manage copying between dev and
release.

(cherry picked from commit 7d6d182c53876898cca9c7ff0244de4a5cf7adb8)

potiuk committed:
Add Pierre to committers (#26609)

(cherry picked from commit fab85f1d7575b939deacfecec13a3f06b7467bc5)

uranusjr committed:
Better @task_group typing powered by ParamSpec and pre-commit (#26366)

(cherry picked from commit 7179eba69e9cb40c4122f3589c5977e536469b13)

potiuk committed:
Remove comon provider docs from ad-hoc releases (#26645)

When we are releasing ad-hoc providers rather than all of them,
we rarely add new stuff (they are mostly bugfixes), so we should
rather not update the "common provider documentation", because
for example when we add (but not yet release) a new provider,
the common indexes might include it and confuse the users.

(cherry picked from commit 3b6176929ccc0351554e5331d920c4ec65b33a38)

potiuk committed:
Do not restart breeze containers after restart (#26647)

In order to avoid initialisation of database and other integrations
when you are entering/leaving breeze, those are started when breeze
starts but not stopped when you leave. Then stopping such running
auxiliary containers should be done with `breeze stop` after you
are done. However those who do not do it, and will restart their
machine will find that the containers get restarted.

This has been added as part of #13446 where health-checks are added.
However "always" was not a good choice. It should have been "on-failure"

(cherry picked from commit 337146dbff6933bd3819ad173f6878ddc934c419)

potiuk committed:
Remove asset compilation for CI image builds (#26665)

The asset compilation is not actually used in CI image so it is not
really needed to be run in CI.

(cherry picked from commit df3bfe3219da340c566afc9602278e2751889c70)

potiuk committed:
Skip sdist providers installation for non-canary builds (#26671)

The sdist providers installation takes a long time and it is unlikely
to fail if wheel installation succeeds. We should skip running the
providers builds for PRs to speed them up. In case there will be
any problem with sdist installation, we will catch it in our
"canary" builds anyway.

(cherry picked from commit e5c903c10e8556ce5bc34d4e8a5289e636750c5c)

potiuk committed:
Make build-images workflow name dynamic (#26690)

As of today you can dynamically name your build-images workflow.
This will be nice change to be able to distinguish the "tests"
from build images workflows.

(cherry picked from commit 2536c00111516a8373b183f5c9173e9771f08326)

rossturk committed:
Update message with new breeze subcommand (#26817)

Signed-off-by: Ross Turk <ross@rossturk.com>

Signed-off-by: Ross Turk <ross@rossturk.com>
(cherry picked from commit 9a6fc73aba75a03b0dd6c700f0f8932f6a474ff7)

potiuk committed:
Switch tests in ci to use Python Breeze (#26612)

This is the last big change in "Rewrite Breeze to Python".

Fixes: #23538
(cherry picked from commit 29217964b678d6af6fbd8c136a06b69f9a215480)

potiuk committed:
Move downgrade/upgrade tests to run new Python breeze (#26828)

This one removes last remaining downgrade/upgrade scripts to use
pure python Breeze.

Thsi also removes the whole "libraries" folder from our repo,
together with a number of common libraries left.

(cherry picked from commit 803d498a2933e446c4301f524b894d1d456a0f5f)

Taragolis committed:
Replace `unittests` test cases by pure `pytest` [Wave-1] (#26831)

(cherry picked from commit 1c7c97630ee226e2dcb3ca90da25dc6a8fbb5b21)

potiuk committed:
Add provider.yaml files to the CI Breeze image check (#26955)

When provider.yaml files change, we are likely to need to rebuild
the CI image - because of the potential dependency changes. This
change triggers warning about it.

When image is rebuild, the provider_dependencies.json file
will be regenerated and new dependencies will be used to build the
image.

(cherry picked from commit d49932375e2087c02939eea15bb65b2487cf3d96)

potiuk committed:
Fix running parallel k8s tests in "depth-first" mode in CI (#26904)

The K8S tests when run in parallel for multiple versions require:

a) vast amount of memory to keep all the cluster running
b) extra time to startup for all k8s tests

This caused a problem when tests in CI were split into separate
steps - because we had to create all clusters before we run tests
even if we limited parallelism.

By using "depth-first" mode, we can run complete tests (with
cluster creation and deletion) in parallel with limited parallelism
without having to create all clusters upfront.

We now add the `run-complete-tests` command that does all the
steps in one function - which makes it much nicer to parallelise
without using too much resources.

There had to be a few fixes done to make it happen.

* Failure in CI for K8S jobs produced really long logs that
  were next to impossible to analyse any problems (for example
  problems involved with lack of memory or with overwhelming K8S
  with too many requests when there were multiple clusters.
  When we combined multiple steps in one output it made the logs
  absolutely unusable.

* Verbose moda has been disabled because it was creating too much
  noise (not that bad as resource dumps, but still lots of it).

* The resource dumps from K8S are now dumped to separate files.
  Those resource dumps can be downloaded separately as they are uploaded
  as artifacts in case of failure.

There were also few fixes to kubernetes tests to make them more stable
and debuggable:

* Assertion which was thrown in "setUp" of K8S test base did not give
  too much information about what the problem was so it has been now
  replaced with a much more user-friendly message showing the actual
  reason (could not check health of the webserver) and
  you can see the root cause alongside the error.

* Few tests in itest_kubernetes_pod_poperator were using fixed path for
  local temp file that caused intermittent failures of the tes when run
  in parallell.

(cherry picked from commit 7bb11a9b762a538f22df58f9fa640ebf837d59e7)

merobi-hub committed:
[DEV] add #protm tag to score calc in protm script (#26605)

* add protm tag to score calc in protm script

Signed-off-by: Michael Robinson <merobi@gmail.com>

* add protm tag search to review comments

Signed-off-by: Michael Robinson <merobi@gmail.com>

* add tag variation to conditions for adding to count

Signed-off-by: Michael Robinson <merobi@gmail.com>

* revise and add a tag variation to conditions for adding to count

Signed-off-by: Michael Robinson <merobi@gmail.com>

* use merged_at date instead of created_at date as cut-ff

Signed-off-by: Michael Robinson <merobi@gmail.com>

* add protm tag to score calc in protm script

Signed-off-by: Michael Robinson <merobi@gmail.com>

* add protm tag search to review comments

Signed-off-by: Michael Robinson <merobi@gmail.com>

* add tag variation to conditions for adding to count

Signed-off-by: Michael Robinson <merobi@gmail.com>

* revise and add a tag variation to conditions for adding to count

Signed-off-by: Michael Robinson <merobi@gmail.com>

* use merged_at date instead of created_at date as cut-ff

Signed-off-by: Michael Robinson <merobi@gmail.com>

* Fix CI

Signed-off-by: Michael Robinson <merobi@gmail.com>
Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>
(cherry picked from commit 0d78ba560dec2e7ea2670744800864906622a4a4)

dstandish committed:
Reduce log level for k8s TCP_KEEPALIVE etc warnings (#26981)

(cherry picked from commit 40d19b9ab89701126653d24d28e477e677f2ea97)

uranusjr committed:
Ask reporter to specify Airflow version (#27004)

(cherry picked from commit d9db89a2d923e1242c32c3c372ddd1c06d58cacb)

ephraimbuddy committed:
Revert "[DEV] add #protm tag to score calc in protm script (#26605)" (#27011)

This reverts commit 0d78ba560dec2e7ea2670744800864906622a4a4.

Reverting due to the static error in main

(cherry picked from commit 7d014b0da5c883eac04148289d236b0f56b2a7be)

merobi-hub committed:
fix bug in protm script (#27015)

Signed-off-by: Michael Robinson <merobi@gmail.com>
(cherry picked from commit 36cc3cae1caf4190b8c7fe8e773886cad04d2712)

ephraimbuddy committed:
Exit on Error in breeze command (#27008)

We changed how we run the DB upgrade/downgrade commands but the new way does not error whenever there's migration error.

Exiting when a command has a returned code of >0 solves the issue

(cherry picked from commit 66fe68103d6b27c613b163b0feb799886751ce56)

o-nikolas committed:
Return the correct error code from breeze shell (#27048)

After #27008 breeze always returned exit code 1 in error cases despite
the actual error code returned by the command. This change plumbs the
return code back to the caller.

(cherry picked from commit 9fd54b7397f81cf71325a98099d44fa33555c612)

kaxil committed:
Listener: Set task on sqlalchemy taskinstance object (#27167)

same as https://github.com/apache/airflow/pull/21157

TaskListener API's contract promises to pass TaskInstance object to listener plugin. However, what happens is not 100% true - the object being passed is one that maps to current SQLAlchemy session.

`_run_raw_task` before merging the TI operates on detached TaskInstance object, then merges it to current session. Since there is no attached object in the SQLAlchemy identity map, SQLAlchemy creates it, and it's this object that's being passed to the SQLAlchemy event listeners.

The problem with that is that when creating new SQLAlchemy object, SQLAlchemy takes care about setting only database-mapped fields. The ones that are purely on the python side, like task aren't being set on the new object.

This PR manually sets `task` on the new SQLAlchemy object, so that `on_task_instance_success` receives proper TaskInstance with task field set.

(cherry picked from commit 395ad7110e53a30a5d33f648d1dd797482eb268c)

phanikumv committed:
Add kerberos environment variables to the docs (#27028)

(cherry picked from commit b8e93bfe64aaaad2a35231b22d8ef1f35f919a74)

BobDu committed:
Fix some bug in web ui dags list page (auto-refresh & jump search null state) (#27141)

* Fix some bug in web ui dags list page

Signed-off-by: BobDu <i@bobdu.cc>

* use switch case & fix remove loading dot

Signed-off-by: BobDu <i@bobdu.cc>

* rename drawDagStats & selectors constants

Signed-off-by: BobDu <i@bobdu.cc>

* Further clean up of duplicate code

Signed-off-by: BobDu <i@bobdu.cc>

Signed-off-by: BobDu <i@bobdu.cc>
(cherry picked from commit ebd34cbd9f62d086bb39727d85fd58a0758edba3)

bhavaniravi committed:
Fix error on running pre-commit without installing breeze (#26985)

* #26984 - Fix error on running pre-commit without installing breeze

* Update dev/breeze/src/airflow_breeze/utils/path_utils.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update dev/breeze/src/airflow_breeze/utils/path_utils.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit ee3625540ff3712bbc6215214e4534d7e91c45fa)

RNHTTR committed:
set executor.job_id to BackfillJob.id for backfills (#27020)

Co-authored-by: Ryan Hatter <ryan.hatter@astronomer.io>
(cherry picked from commit 57bb54d458aec4bea6977b2e410107df4c281de6)

potiuk committed:
Add better diagnostics on auto dependencies upgrade (#27215)

Whenever upgrade-to-newer-dependencies is set, our CI will attempt
to upgrade all dependencies to latest versions released. However so
far it was not clear which dependencies were upgraded so if CI
failed because of some dependency release, you had to guess and
manually check which dependencies got upgraded.

This PR adds a step that will show the diff between the constraints
in the target branch (`constraints-main` or `constraints-X-Y` in
case upgrade-to-newer-dependencies is enabled.

(cherry picked from commit 8f5ed4b3d1aec4507b8f7869005296ba06512cc2)

potiuk committed:
Fix constraint upgrade (#27221)

It turnded out that #27215 has a problem when runnning when
in-workflow-build is False - additional condition has to be added for
it.

Additionally, the constraint generation is run for all python
versions in parallel instaed of running only for default python
version.

(cherry picked from commit e789a2ba9fbcc0a2ac8a865f39fbb488c23e64f7)

rlukin committed:
Fix typo in set-up-database.rst (#27137)

(cherry picked from commit 131d339696e9568a2a2dc55c2a6963897cdc82b7)

potiuk committed:
Make RotatingFilehandler used in DagProcessor non-caching (#27223)

The RotatingFileHandler is used when you enable it via
`CONFIG_PROCESSOR_MANAGER_LOGGER=True` and it exhibits similar
behaviour as the FileHandler had when it comes to caching
the file on the Kernel level. While it is harmless (the cache
will be freed when needed), it is also misleading for those who
are trying to understand memory usage by Airlfow.

The fix is to add a custom non-caching RotatingFileHandler similarly
as in #18054.

Note that it will require to manually modify local settings if
the settings were created before this change.

Fixes: #27065
(cherry picked from commit 126b7b8a073f75096d24378ffd749ce166267826)

potiuk committed:
Upgrade `pip' to 22.3 (#27203)

(cherry picked from commit 78669ffcfd210a6e5f802e2c7b09cfdc4bb6a243)

potiuk committed:
Update Node.js in Github Actions to Node 16 (#27192)

We were using an old version of some actions  that used Node 12
and node 12 has been deprecated. This PR bumps all necessary actions
to use Node 16.

(cherry picked from commit c3646af70a72eae4e5ffbf2d9c246d5eaac8571e)

bbovenzi committed:
reduce extraneous task log requests (#27233)

(cherry picked from commit e73e90e388f7916ae5eea48ba39687d99f7a94b1)

dylan-turnbull committed:
Update dags.rst (#27110)

Corrects the use of an incorrect word when explaining that pickling can't be enabled for serialization.

(cherry picked from commit 7fcdb490b4932e55fd0d92443e7ade451802a564)

potiuk committed:
Add dynamic attr check for ast.unparse method in pre-commit (#27280)

(cherry picked from commit ced2925fc31e6516c9e7163f60cfb17890a0a1a0)

ephraimbuddy committed:
Fix IntegrityError during webserver startup (#27297)

(cherry picked from commit 8f99c793ec4289f7fc28d890b6c2887f0951e09b)

pigidser committed:
Add missed import in the Trigger Rules example (#27309)

* Add missed import in the Trigger Rules example

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 426f527690f7713bc1e9a48b3c91a659d7b338bc)

notatallshaw-gts committed:
Update SLA wording to reflect it is relative to Dag Run start. (#27111)

(cherry picked from commit 639210a7e0bfc3f04f28c7d7278292d2cae7234b)

bmtKIA6 committed:
Fix typo (#27327)

(cherry picked from commit 9c73b3f7fc1d18925d0ed09e8719f53b8147b0f2)

potiuk committed:
fIx failing masking tests for python < 3.10 (#27337)

Seems that the number of times user is printed in stack trace
depend on Python version. The fix in #27335 seems to only have
worked for Python 3.10 with the 1.0.0 of exceptiongroup the
stack trace has less stack levels.

(cherry picked from commit fcec804ff513d8f148d5069a48e7adf703855a96)

potiuk committed:
fIx failing masking tests for python < 3.10 (#27337)

Seems that the number of times user is printed in stack trace
depend on Python version. The fix in #27335 seems to only have
worked for Python 3.10 with the 1.0.0 of exceptiongroup the
stack trace has less stack levels.

(cherry picked from commit fcec804ff513d8f148d5069a48e7adf703855a96)

potiuk committed:
Fix caching in CI for pre-commits (#27368)

For some time, caching in pre-commits was broken and some of the
pre-commit installation was unnecessary. There were few missing
or mis-sequenced steps when the CI cache was declered and it made
regular PRs us a bit (1-2 minutes) longer time for pre-commit
setup than it was needed. In case of image building, there is no
need to setup all pre-commits because only one pre-commit is
needed (update-generated dependencies) - instead we can simply
install rich and yaml locally and run the pre-comit script directly.

This PR cleans it up.

(cherry picked from commit 62a870bbdaf1cb1b572496e90cf7e9418974870b)

enisnazif committed:
Fix python external template keys (#27256)

(cherry picked from commit 27a92fecc9be30c9b1268beb60db44d2c7b3628f)

ephraimbuddy committed:
Add case insensitive constraint to username (#27266)

This helps us to recognize usernames properly

(cherry picked from commit 1d25105189db09df53327240a01f39e09e10b15a)

ecerulm committed:
Fix BaseOperator link (#27441)

(cherry picked from commit 8829e999077040248ab302faa76897915430bfd3)

jedcunningham committed:
Fix behavior of `_` when searching for DAGs (#27448)

On the homepage, when you search for DAGs, `_` wasn't behaving as most
people might expect since it is a single character wildcard in SQL. We
will automatically escape it now.

(cherry picked from commit eda089334864547e18b71cb22fa231872edd4e8b)

jedcunningham committed:
Document that KubernetesExecutor overwrites container args (#27450)

The executor will overwrite the args for the container so that the
worker runs the right task.

(cherry picked from commit 8f5e100f30764e7b1818a336feaa8bb390cbb327)

csm10495 committed:
SLAMiss is nullable and not always given back when pulling task instances (#27423)

* SLAMiss is nullable and not always given back when pulling task instances

See https://github.com/apache/airflow-client-python/issues/50 for an example of the related error

* SLAMiss is nullable and not always given back when pulling task instances

See https://github.com/apache/airflow-client-python/issues/50 for an example of the related error

(cherry picked from commit 63638cd2162219bd0a67caface4b1f1f8b88cc10)

SnowOracle committed:
Fix wording in DAG Runs description(#27470)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 104ce1e7d81457a768366bd1ba319ecad934a921)

umren committed:
Update index.rst (#27529)

docker-compose doesn'to work, we need to use docker compose now

(cherry picked from commit 2a4f81c1782044f3e9feb2dc968e56ebc0525a1a)

potiuk committed:
Increase time for constraints generation (#27534)

In case we have some new versions of dependencies released, generation
of the constraints (especially for PyPI-released providers) might
take more time than usual. Increasing time for the constraint generation
will help to handle those cases (note that once constraint generation
succeeds, the constraints will be committed to repository and next time
constraint generation should be faster.

(cherry picked from commit 194763867cab14b87f62d8ca6a818eac5fab9fd2)

potiuk committed:
Add azure, google, authentication library limits to eaager upgrade (#27535)

When generating constraints for released providers we hit the same
`pip` resolver backtracking issue - where it took very, very long
time to resolve the dependencies for installing released providers
on latest airflow from sources.

In #27531 we've added limits to provider.yaml but currently released
providers do not have those limits, so until we release them, we
should add the limits to "eager upgrade dependencies" in our CI
image - to help `pip` to figure out the right set of dependencies
much faster.

Also fixed a problem with extra "." in adal specification (seems it's
been ignored by `pip` anyway)

(cherry picked from commit a50195d617ca7c85d56b1c138f46451bc7599618)

ephraimbuddy committed:
Fix sqlalchemy primary key black-out error on DDRQ (#27538)

closes https://github.com/apache/airflow/issues/27509

(cherry picked from commit fc59b02cfac7fd691602edc92a7abac38ed51531)

potiuk committed:
Synchronize eager upgrade limits in all places where they are used (#27539)

(cherry picked from commit 6307dc1acd0d8dcfc206c7980ede1f3d26fb5984)

jacobm3 committed:
Correct timer units to seconds from milliseconds. (#27360)

``dagrun.duration.success.<dag_id>``
``dagrun.schedule_delay.<dag_id>``
``dagrun.<dag_id>.first_task_scheduling_delay`` are reported to statsd in seconds, not milliseconds. Other timers in this list might be suspect, but I haven't checked them yet.

(cherry picked from commit 25d7ca918e2d382cccd7d70386a1055d33bb517b)

jordanjeremy committed:
Fix getting the dag/task ids from base executor (#27550)

(cherry picked from commit f304df7857eea344a7c274dcf41bbe5271a0083f)

dependabot[bot] committed:
Bump loader-utils from 1.4.0 to 1.4.1 in /airflow/www (#27552)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.0 to 1.4.1.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.1/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.0...v1.4.1)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit 9936d61f7b1e4e122bdbdf099df75c48aa9bc033)

ephraimbuddy committed:
Fix mini scheduler expansion of mapped task  (#27506)

We have a case where the mini scheduler tries to expand a mapped task even when the downstream tasks are not yet done.

The mini scheduler extracts a partial subset of a dag and in the process, some upstream tasks are dropped.
If the task happens to be a mapped task, the expansion will fail since it needs the upstream output to make the expansion. When the expansion fails, the task is marked as `upstream_failed`. This leads to other downstream tasks being marked as upstream failed.

The solution was to ignore this error and not mark the mapped task as upstream_failed when the expansion fails and the dag is a partial subset

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
(cherry picked from commit ed92e5d521f958642615b038ec13068b527db1c4)

ephraimbuddy committed:
Update version to 2.4.3

treyyi committed:
Remove self.using_mysql attribute (#27571)

This is to remove `self.using_mysql` attribute since it is not used in scheduler_job.py.

(cherry picked from commit 2ac45b011d04ac141a15abfc24cc054687cadbc2)

jedcunningham committed:
Replace FAB url filtering function with Airflows (#27576)

Use our url filtering util function so there is consistency between
FAB and Airflow routes.

(cherry picked from commit b33d22c2243c552f4d1372bbe6b75e5fb097f567)

ephraimbuddy committed:
Add release notes

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- ``SLAMiss`` is nullable and not always given back when pulling task instances (#27423)
+- Fix behavior of ``_`` when searching for DAGs (#27448)
+- Fix getting the ``dag/task`` ids from base executor (#27550)
+- Fix sqlalchemy primary key black-out error on DDRQ (#27538)
```
```suggestion
- Fix SQLAlchemy primary key black-out error on DDRQ (#27538)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Fix IntegrityError during webserver startup (#27297)
+- Add case insensitive constraint to username (#27266)
+- Fix python external template keys (#27256)
+- reduce extraneous task log requests (#27233)
```
```suggestion
- Reduce extraneous task log requests (#27233)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Fix python external template keys (#27256)
+- reduce extraneous task log requests (#27233)
+- Make ``RotatingFilehandler`` used in ``DagProcessor`` non-caching (#27223)
+- Listener: Set task on sqlalchemy taskinstance object (#27167)
```
```suggestion
- Listener: Set task on SQLAlchemy TaskInstance object (#27167)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- reduce extraneous task log requests (#27233)
+- Make ``RotatingFilehandler`` used in ``DagProcessor`` non-caching (#27223)
+- Listener: Set task on sqlalchemy taskinstance object (#27167)
+- Fix some bug in web ui dags list page (auto-refresh & jump search null state) (#27141)
```
```suggestion
- Fix dags list page auto-refresh & jump search null state (#27141)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Make ``RotatingFilehandler`` used in ``DagProcessor`` non-caching (#27223)
+- Listener: Set task on sqlalchemy taskinstance object (#27167)
+- Fix some bug in web ui dags list page (auto-refresh & jump search null state) (#27141)
+- set ``executor.job_id`` to ``BackfillJob.id`` for backfills (#27020)
```
```suggestion
- Set ``executor.job_id`` to ``BackfillJob.id`` for backfills (#27020)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+
+Bug Fixes
+^^^^^^^^^
+- Replace FAB url filtering function with Airflow (#27576)
```
```suggestion
- Replace FAB url filtering function with Airflow's (#27576)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+Doc only changes
+^^^^^^^^^^^^^^^^
+- Update ``index.rst`` (#27529)
+- Fix wording in DAG Runs description(#27470)
```
```suggestion
- Fix wording in DAG Runs description (#27470)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+
+Doc only changes
+^^^^^^^^^^^^^^^^
+- Update ``index.rst`` (#27529)
```
```suggestion
- Use correct executable in docker compose docs (#27529)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Fix ``BaseOperator`` link (#27441)
+- Correct timer units to seconds from milliseconds. (#27360)
+- Add missed import in the Trigger Rules example (#27309)
+- Fix typo in ``set-up-database.rst`` (#27137)
```
```suggestion
```
Can probably skip this one.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Add missed import in the Trigger Rules example (#27309)
+- Fix typo in ``set-up-database.rst`` (#27137)
+- Update SLA wording to reflect it is relative to ``Dag Run`` start. (#27111)
+- Update ``dags.rst`` (#27110)
```
```suggestion
```

Typo, we can skip this one too.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Fix mini scheduler expansion of mapped task  (#27506)
+- ``SLAMiss`` is nullable and not always given back when pulling task instances (#27423)
+- Fix behavior of ``_`` when searching for DAGs (#27448)
+- Fix getting the ``dag/task`` ids from base executor (#27550)
```
```suggestion
- Fix getting the ``dag/task`` ids from BaseExecutor (#27550)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
+- Update ``index.rst`` (#27529)
+- Fix wording in DAG Runs description(#27470)
+- Document that ``KubernetesExecutor`` overwrites container args (#27450)
+- Fix ``BaseOperator`` link (#27441)
```
```suggestion
- Fix ``BaseOperator`` links (#27441)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,53 @@
 
 .. towncrier release notes start
 
+Airflow 2.4.3 (2022-11-13)
```
```suggestion
Airflow 2.4.3 (2022-11-14)
```

Maybe target Monday instead?

jedcunningham committed:
Init all Airflow routes when testing API (#27586)

We should init all of the Airflow routes when testing the API, as we now
have a code path that expects them.

(cherry picked from commit b423869440088d46d2313065d4abc558ce5f0cad)

ashb committed:
Fix double logging with some task logging handler (#27591)

A previous change to fix disappearing log messages turned on propagation
for the `airflow.task` logger, and disabled it again after
`set_context()` was called, but only if that function returned a special
sentinel value.

For the "core" task log handlers we returned them, but some providers
weren't "correctly" subclassing and weren't returning this sentinel
value.

The fix here is to change the logic from disable only on special value
to disable by default and maintain propagation on special value; this
means that if a handler doesn't return the value from `super()` (or if
they don't even subclass the default handler) propagation will still be
disabled by default.

(cherry picked from commit 933fefca27a5cd514c9083040344a866c7f517db)

potiuk committed:
Fix test hooks also for Azure Batch (#27607)

The #27601 also needs fix in Hook test cases.

(cherry picked from commit 3e3ff7ce5232a2662cd9f35e21ee8f088e568d6d)

potiuk committed:
Only run mypy/flake static tests for providers in main branch (#27608)

(cherry picked from commit 93459535684b847f16c4fdeb043aeb10dae1c852)
